jetty-ee10/jetty-ee10-tests/jetty-ee10-test-client-transports/src/test/java/org/eclipse/jetty/ee10/test/client/transport/RequestReaderTest.java

```diff
@@ -15,23 +15,199 @@
+    {
+        assumeTrue(transport.isMultiplexed());
+
+        CountDownLatch servletDoneLatch = new CountDownLatch(1);
+        AtomicReference<Throwable> serverError = new AtomicReference<>();
+        CountDownLatch errorDoneLatch = new CountDownLatch(1);
+        start(transport, new HttpServlet()
+        {
+            @Override
+            protected void service(HttpServletRequest request, HttpServletResponse response) throws IOException
+            {
+                AsyncContext asyncContext = request.startAsync();
+                asyncContext.addListener(new AsyncListener()
+                {
+                    @Override
+                    public void onComplete(AsyncEvent event)
+                    {
+                    }
+
+                    @Override
+                    public void onTimeout(AsyncEvent event)
+                    {
+                    }
+
+                    @Override
+                    public void onError(AsyncEvent event) throws IOException
+                    {
+                        serverError.set(event.getThrowable());
+                        errorDoneLatch.countDown();
+                        response.sendError(567);
+                        asyncContext.complete();
+                    }
+
+                    @Override
+                    public void onStartAsync(AsyncEvent event)
+                    {
+                    }
+                });
+
+                ServletInputStream inputStream = request.getInputStream();
+                inputStream.setReadListener(new ReadListener()
+                {
+                    @Override
+                    public void onDataAvailable()
+                    {
+                        servletDoneLatch.countDown();
+                    }
+
+                    @Override
+                    public void onAllDataRead()
+                    {
+                    }
+
+                    @Override
+                    public void onError(Throwable t)
+                    {
+                    }
+                });
+            }
+        });
+
+        AtomicReference<Result> resultRef = new AtomicReference<>();
+        try (AsyncRequestContent content = new AsyncRequestContent(ByteBuffer.allocate(16)))
+        {
+            Request request = client.newRequest(newURI(transport))
+                .method("POST")
+                .timeout(5, TimeUnit.SECONDS)
+                .body(content);
+            request.send(resultRef::set);
+            servletDoneLatch.await();
+            request.abort(new ArithmeticException());
+        }
+
+        assertTrue(errorDoneLatch.await(5, TimeUnit.SECONDS));
+        assertThat(serverError.get(), instanceOf(EofException.class));
+
+        await().pollDelay(1, TimeUnit.MILLISECONDS).atMost(5, TimeUnit.SECONDS).until(resultRef::get, not(nullValue()));
+        Result result = resultRef.get();
+        assertThat(result.getFailure(), instanceOf(ArithmeticException.class));
```

Comment:
Can you explicitly assert whether you expect a 567 response or not?

Reply:
I've added an assertion for the status code.